### PR TITLE
Offline Mode: Add conflict resolution dialog

### LINKED ISSUE #22583: Offline Mode: Resolve Conflict
## Tasks

- [x] Enable the save button after a selection has been made
- [x] Only single selection is allowed

### PR DESCRIPTION
**Pull request by momo-ozawa**
Fixes #22583

## Description
- Adds a conflict resolution dialog that consolidates changes if the server returns a 409 (ref: pcdRpT-63f-p2)

### Pick local

- Publish a post
- Open the post in the app editor and the web editor
- Make a change on the web and click "Update"
- Make a change on the app and tap "Update"
- ✅ Verify: An error is displayed
- Tap "OK"
- ✅ Verify: "Save" button is enabled if a selection has been made; only single selection is supported
- Select "Current device" and tap "Save"
- Refresh web editor
- ✅ Verify: web editor reflects changes made on the app

https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/2e065287-f397-4b79-a310-6cdd25f35ce8

### Pick remote

- Publish a post
- Open the post in the app editor and the web editor
- Make a change on the web and click "Update"
- Make a change on the app and tap "Update"
- ✅ Verify: An error is displayed
- Tap "OK"
- ✅ Verify: "Save" button is enabled if a selection has been made; only single selection is supported
- Select "Another device" and tap "Save"
- ✅ Verify: app editor reflects changes made on the web

https://github.com/wordpress-mobile/WordPress-iOS/assets/6711616/3347bc4b-df98-4634-8068-5fb8f4b08a17

## Regression Notes
1. Potential unintended areas of impact
n/a

2. What I did to test those areas of impact (or what existing automated tests I relied on)
n/a

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

Testing checklist:
- [ ] WordPress.com sites and self-hosted Jetpack sites.
- [ ] Portrait and landscape orientations.
- [ ] Light and dark modes.
- [ ] Fonts: Larger, smaller and bold text.
- [ ] High contrast.
- [ ] VoiceOver.
- [ ] Languages with large words or with letters/accents not frequently used in English.
- [ ] Right-to-left languages. (Even if translation isn’t complete, formatting should still respect the right-to-left layout)
- [ ] iPhone and iPad. 
- [ ] Multi-tasking: Split view and Slide over. (iPad)

### CODE REVIEW

File: WordPress/WordPress.xcodeproj/project.pbxproj [hunk 1]
```diff
@@ -21008,11 +21016,11 @@
 			files = (
 			);
 			inputPaths = (
-				"$SRCROOT/../Scripts/BuildPhases/CopyGutenbergJS.inputs.xcfilelist",
+				$SRCROOT/../Scripts/BuildPhases/CopyGutenbergJS.inputs.xcfilelist,
```
Comment (kean) [inline WordPress/WordPress.xcodeproj/project.pbxproj]: 👀

**Comment by kean** [comment]
Works as expected.

There is one more entry point that we need to cover:

Screenshot 2024-04-08 at 8 30 13 AM

> if you tap on the post or the error indicator, it shows the new “Resolve Conflict” screen

**Review by kean — APPROVED** [review]

File: WordPress/WordPress.xcodeproj/project.pbxproj (hunk 1 above)
Comment (momo-ozawa) [reply WordPress/WordPress.xcodeproj/project.pbxproj]: Weird. Xcode must have applied some changes. I undid them https://github.com/wordpress-mobile/WordPress-iOS/pull/22879/commits/39a538b6298130fe9fae6e711aa80d47c7ecfc3a

**Comment by momo-ozawa** [comment]
> if you tap on the post or the error indicator, it shows the new “Resolve Conflict” screen

Sounds good, will open a new PR

**Pull request merged**